java/sage/media/format/FormatParser.java

```diff
@@ -395,6 +420,38 @@ private static boolean addAdditionalMetadata(java.io.File f, ContainerFormat for
     return rv;
   }
 
+  private static FormatParserPlugin getFormatParserPluginInstance()
+  {
+    /*
+     * Check to see if an instance has already been created.  If not check to see if one is configured and attempt to create an instance
+    */
+    if(FormatParser.formatParserPluginInstance == null)
+    {
+      String parsePlugin = sage.Sage.get("mediafile_mediaformat_parser_plugin", "");
+      
+      if(parsePlugin.isEmpty())
+      {
+        FormatParser.formatParserPluginInstance = null;
+      }
+      else
+      {
+        try
```

Comment:
This isn't thread-safe, so either put a static lock in here...or put a comment that explains we know that it's not thread-safe and it's OK for one plugin instance to clobber another if there's a race condition.

Reply:
Would be OK if I made the method synchronized?  I would think this would not get called by more than one thread very often.